client/src/components/CourseSearchBar.tsx

```diff
@@ -17,6 +17,22 @@ export const CourseSearchBar = ({
   const parser = new DOMParser();
 
   const [searchSelected, setSearchSelected] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'ArrowUp') {
+      event.preventDefault();
+      setSelectedIndex((prevIndex) => (prevIndex > 0 ? prevIndex - 1 : -1));
+    } else if (event.key === 'ArrowDown') {
+      event.preventDefault();
+      setSelectedIndex((prevIndex) =>
+        prevIndex < results.length - 1 ? prevIndex + 1 : results.length - 1
+      );
+    }
+    if (selectedIndex > -1 && event.key === 'Enter') {
+      window.location.href = `/course/${results[selectedIndex]._id}`;
```

Comment:
I think we should use `useNavigate` from react router here instead of just changing the window location ourselves